File: 0001/go/solution.go
package main

import "fmt"

func calculate_sum(n int) int {
    var i int = 0;
    var sum int = 0

    for ; i < n; i++ {
        if i % 3 == 0 || i % 5 == 0 {
            sum = sum + i
        }
    }
    return sum;
}

func main() {
    fmt.Println(calculate_sum(1000))
}

File: 0001/rust/solution.rs
fn calculate_sum(limit: u32) -> u32
{
    let mut i: u32 = 0;
    let mut sum: u32 = 0;
    
    while i < limit {
        if i % 3 == 0 || i % 5 == 0 {
            sum = sum + i;
        }
        i += 1;
    }
    return sum;
}

fn main()
{
    let limit: u32 = 1000;
    println!("{}", calculate_sum(limit));
}
